### DISCUSSION TITLE
Docs: Docs for all (previous) major versions

### DISCUSSION THREAD

**Discussion by tasawar-hussain** [vercel/next.js#66000] [Ideas]
### What is the update you wish to see?

There is a select menu on the Nextjs website, using which we can switch to the previous version of Nextjs docs.

### Is there any context that might help us understand?

Not all projects use the latest version of Next JS, and if there is a need to consult the documentation from the previous version, there is no such thing.

**Comment by samcx** [comment]
@tasawar-hussain Hi there!

To clarify, we want to keep any <picture data-single-emoji=":docs:" title=":docs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/docs/172e1b9b7d137b8a.png" alt=":docs:" align="absmiddle"></picture> feature requests in <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions)—I will proceed with doing that!

**Comment by sos0** [comment]
Bumping and reposting this [issue](https://github.com/vercel/next.js/issues/66357) that got closed by Delba.

-----

## What is the update you wish to see?

Since Next.js is constantly being updated at breakneck speed, it'd be immensely helpful for developers if we had "versioned documentation". For example, documentation corresponding to Next 12, Next 13, Next 14, and Next 15 can be separately made available for developers, alongside recommendations to upgrade to certain versions, just as the Pages Router documentation includes a recommendation to migrate to the App Router.

## Is there any context that might help us understand?

Considering not all developers choose to migrate to latest/newest versions of Next.js, "versioned documentation" would be the ideal source of information relevant to each developer. A developer utilizing a lower versioned product can no longer rely on the current documentation since the current documentation may contain new features incompatible with old versions.
